src/osd/OSD.cc

```diff
@@ -5536,6 +5543,27 @@ void OSD::_send_boot()
   set_state(STATE_BOOTING);
 }
 
+std::string OSD::_collect_compression_algorithms()
+{
+  using std::experimental::make_ostream_joiner;
+
+  const auto& compression_algorithms = Compressor::compression_algorithms;
+  const auto& plugin_registry = cct->get_plugin_registry()->plugins;
+
+  if (plugin_registry.empty())
+   return {};
+
+  ostringstream os;
+
+  copy_if(begin(compression_algorithms), end(compression_algorithms),
+          make_ostream_joiner(os, ", "),
```

Comment:
it looks like this will print pairs instead of just the names

Reply:
#21809